cli/dcoscli/main.py

```diff
@@ -67,17 +64,31 @@ def _main():
             msg = 'Unable to get the cluster_id of the cluster.'
             logger.exception(msg)
 
+    # send args call to segment.io
+    reporting_executor = ThreadPoolExecutor(max_workers=2)
+    report = config.get('core.reporting', True)
+    if report and dcoscli.version != 'SNAPSHOT':
+        analytics.segment_track_cli(reporting_executor, config, cluster_id)
+
     # the call to retrieve cluster_id must happen before we run the subcommand
     # so that if you have auth enabled we don't ask for user/pass multiple
     # times (with the text being out of order) before we can cache the auth
     # token
-    subproc = Popen([executable,  command] + args['<args>'],
-                    stderr=PIPE)
-
-    if dcoscli.version != 'SNAPSHOT':
-        return analytics.wait_and_track(subproc, cluster_id)
+    if command in subcommand.default_subcommands():
+        sys.argv[1:] = [command] + args['<args>']
+        sc = subcommand.SubcommandMain(command)
     else:
-        return analytics.wait_and_capture(subproc)[0]
+        executable = subcommand.command_executables(command)
+        sc = subcommand.SubcommandProcess(executable, command, args['<args>'])
+
+    exitcode, err = sc.run_and_capture()
+
+    if err:
+        if dcoscli.version != 'SNAPSHOT' and report:
```

Comment:
I see that we do this in a couple of places. I think we should move this check into the `analytics` module.